### PR TITLE
SettingsHolder can be cleared

### PR DESCRIPTION
**Pull request by danielmarbach**
@bording That would be my idea to https://github.com/Particular/NServiceBus/pull/3428

@Particular/nservicebus-maintainers Thoughts?

### CODE REVIEW

File: src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt [hunk 1]
```diff
@@ -2507,6 +2507,7 @@ namespace NServiceBus.Settings
         public SettingsHolder() { }
         public void ApplyTo<T>(NServiceBus.ObjectBuilder.IComponentConfig config) { }
         public void ApplyTo(System.Type componentType, NServiceBus.ObjectBuilder.IComponentConfig config) { }
+        public void Clear() { }
```
Comment (yvesgoeleven) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: Isn't it dangerous to make this public? We really don't want users to call this at runtime?

Comment (danielmarbach) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: Good point, I'll change it

Comment (andreasohlund) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: We only expose the IReadOnlySettings at runtime (unless the user is smart enough cast it to `SettingHolder`)

Comment (danielmarbach) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: Should I still make it internal or leave as is?

Comment (danielmarbach) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: I'm Ok with that approach

**Pull request merged**